src/App.tsx

```diff
@@ -1,8 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import {
+  getData,
+  PostsURL,
+  UsersURL,
+  CommentsURL,
+} from './api';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import {
+  FullPost,
+  Post,
+  User,
+  CommentInterface,
+} from './Interfaces/Interfaces';
+import { PostsList } from './components/PostsList/PostsList';
+
+const App: React.FC = () => {
+  const [fullPosts, setFullPosts] = useState<FullPost[]>([]);
+  const [isLoading, switchIsLoading] = useState<boolean>(false);
+
+  const getFullPosts = async () => {
+    switchIsLoading(true);
+    const users = await getData<User[]>(UsersURL);
+    const posts = await getData<Post[]>(PostsURL);
+    const comments = await getData<CommentInterface[]>(CommentsURL);
+    const preperedPosts = posts.map((post) => ({
+      ...post,
+      user: users.find((user) => user.id === post.userId) as unknown as User,
```

Comment:
```suggestion
      user: users.find((user) => user.id === post.userId) as unknown as User,
```
зачем тебе as unknown ?